### PR TITLE
Update transaction panel UI for Filecoin transactions

### PR DESCRIPTION
**Pull request by muliswilliam**
Update panel transaction UI for Filecoin transactions

Resolves https://github.com/brave/brave-browser/issues/22020, https://github.com/brave/brave-browser/issues/23192

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
https://user-images.githubusercontent.com/48117473/172464369-34096180-fc53-4b71-8bf4-ba67092de2d9.mov

### CODE REVIEW

File: components/brave_wallet_ui/common/hooks/transaction-parser.ts [hunk 1]
```diff
@@ -99,21 +101,24 @@ export function useTransactionFeesParser (selectedNetwork: BraveWallet.NetworkIn
...
+    const isFilTransaction = filTxData !== undefined
+    const gasLimit = isFilTransaction ? filTxData.gasLimit : txData?.baseData.gasLimit || ''
     const gasPrice = txData?.baseData.gasPrice || ''
     const maxFeePerGas = txData?.maxFeePerGas || ''
     const maxPriorityFeePerGas = txData?.maxPriorityFeePerGas || ''
     const isEIP1559Transaction = maxPriorityFeePerGas !== '' && maxFeePerGas !== ''
-    const gasFee = isSolTransaction
-      ? new Amount(solFeeEstimates?.fee.toString() ?? '').format()
-      : isEIP1559Transaction
+    let gasFee: string
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/hooks/transaction-parser.ts:110]: Let's keep this as a `const`

File: components/brave_wallet_ui/common/hooks/transaction-parser.ts [hunk 2]
```diff
@@ -202,21 +209,32 @@ export function useTransactionParser (
...
-    const value = isSPLTransaction
-      ? solTxData?.amount.toString() ?? ''
-      : isSolTransaction
-        ? solTxData?.lamports.toString() ?? ''
-        : txData?.baseData.value || ''
-    const to = isSolTransaction
-      ? solTxData?.toWalletAddress ?? ''
-      : txData?.baseData.to || ''
+
+    let value = ''
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/hooks/transaction-parser.ts:221]: Lets keep the ternary and this as a const

File: components/brave_wallet_ui/common/hooks/transaction-parser.ts [hunk 3]
```diff
@@ -202,21 +209,32 @@ export function useTransactionParser (
...
+    if (isSolTransaction) {
+      value = solTxData?.amount.toString() ?? ''
+    } else if (isSolTransaction) {
+      value = solTxData?.lamports.toString() ?? ''
+    } else if (isFilTransaction) {
+      value = filTxData.value || ''
+    }
+
+    let to = ''
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/hooks/transaction-parser.ts:230]: keep as const

**Review by josheleonard — CHANGES_REQUESTED** [review]

**Review by josheleonard — APPROVED** [review]
LGTM

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.41.50 Chromium: 102.0.5005.99 (Official Build) nightly (64-bit) 
-- | --
Revision | `870f7bcc58dfa811cc68c2186439721385e086d0-refs/branch-heads/5005@{#1125}`
OS | Windows 11 Version 21H2 (Build 22000.708)

- Verified steps from PR
- Verified transaction panel shows UI for gas and fee
- Verified panel shows sending and receiving address
- Verified selecting transaction hash loads  the correct block explorer

https://user-images.githubusercontent.com/17010094/172908592-b7c74cbb-f829-48a1-8c09-5e3d4b51e7e7.mp4

**Pull request merged**